share/picongpu/unit/radiationFormFactor.cpp

```diff
@@ -0,0 +1,184 @@
+ */
+namespace picongpu
+{
+    constexpr picongpu::float_X SPEED_OF_LIGHT = 1.0;
+    constexpr picongpu::float_X CELL_WIDTH = 2.0;
+    constexpr picongpu::float_X CELL_HEIGHT = 3.0;
+    constexpr picongpu::float_X CELL_DEPTH = 4.0;
+    constexpr picongpu::float_X DELTA_T = 5.0;
+
+    PMACC_CONST_VECTOR(picongpu::float_X, TEST_DIM, cellSize, CELL_WIDTH, CELL_HEIGHT, CELL_DEPTH);
+} // namespace picongpu
+
+#include <picongpu/plugins/radiation/VectorTypes.hpp>
+#include <picongpu/plugins/radiation/radFormFactor.hpp>
+#include <picongpu/plugins/radiation/vector.hpp>
+
+
+//! Helper to setup the PMacc environment
+using TestFixture = pmacc::test::PMaccFixture<TEST_DIM>;
+static TestFixture fixture;
+
+using namespace picongpu;
+using namespace pmacc;
+
+constexpr uint32_t numValues = 1024;
+constexpr uint32_t elemPerBlock = 256;
+
+/** check if floating point result is equal
+ *
+ * Allows an error of one epsilon.
+ * @return true if equal, else false
+ */
+template<typename T>
+static bool isApproxEqual(T const& a, T const& b)
+{
+    return a == Catch::Approx(b).margin(std::numeric_limits<T>::epsilon());
+}
+
+/** Test radiation form factor function
+ *
+ * Compares the on host and on device computed result.
+ *
+ * @tparam T_FormFactorFunctions radiation window function
+ */
+template<typename T_FormFactorFunctions = boost::mpl::_1>
+struct TestFormFactor
+{
+    /** Validates the shape
+     *
+     * @param inOmegaBuffer Buffer with omega [0.0;Inf)
+     * @param inMacroWeightingBuffer Buffer with makro particle weighting [0.0;Inf)
+     */
+    template<typename T_OmegaBuffer, typename T_MacroWeightingBUffer>
+    void operator()(T_OmegaBuffer& inOmegaBuffer, T_MacroWeightingBUffer& inMacroWeightingBuffer)
+    {
+        std::cout << "Test form factor function" << typeid(T_FormFactorFunctions).name() << std::endl;
+        ::pmacc::DeviceBuffer<float_X, 1u> deviceInOmegaBuffer(numValues);
+        ::pmacc::DeviceBuffer<float_X, 1u> deviceInMacroWeightingBuffer(numValues);
+
+        ::pmacc::HostBuffer<float_X, 1u> resultHost(numValues);
+        ::pmacc::DeviceBuffer<float_X, 1u> resultDevice(numValues);
+
+        deviceInOmegaBuffer.copyFrom(inOmegaBuffer);
+        deviceInMacroWeightingBuffer.copyFrom(inMacroWeightingBuffer);
+
+        resultDevice.setValue(0.0_X);
+
+        auto shapeTestKernel
+            = [this] DEVICEONLY(auto const& worker, auto const& omega, auto const& weighting, auto result)
+        {
+            auto blockIdx = worker.blockDomIdxND().x();
+
+            auto forEach = lockstep::makeForEach<elemPerBlock>(worker);
+
+            forEach(
+                [&](uint32_t const idx)
+                {
+                    auto valueIdx = blockIdx * elemPerBlock + idx;
+
+                    if(valueIdx < numValues)
+                    {
+                        float_X const omg = omega[valueIdx];
+                        auto const windowFunc = T_FormFactorFunctions{omg, {1.0, 1.0, 1.0}};
```

Comment:
```suggestion
                        auto const formFactorFunc = T_FormFactorFunctions{omg, {1.0, 1.0, 1.0}};
```

Reply:
Furthermore `{1.0, 1.0, 1.0}` is not a unit vector (of length 1) and thus will create nonsense. 
How about 
```c++
std::uniform_real_distribution<> phi_dist(0.0, 2.0*PI);
std::uniform_real_distribution<> theta_dist(0.0, 1.0*PI);
...
... {math::cos(phi) * math::sin(theta), math::sin/phi) * math::sin(theta), math::cos(theta)} ...
```